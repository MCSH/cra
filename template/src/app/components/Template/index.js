/**
 *
 * Template
 *
 */
import React from 'react';
import { Helmet } from 'react-helmet-async';

function Template(props = {}) {
  const title = props.title || "";
  const description = props.description || "";
  return (
    <>
      <Helmet>
        <title>{title}</title>
        <meta name="description" content={description} />
      </Helmet>
      {props.children}
    </>
  );
}

export default Template;
