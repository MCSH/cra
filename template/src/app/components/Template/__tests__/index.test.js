import React from 'react';
import { render } from '@testing-library/react';
import { Store } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { HelmetProvider } from 'react-helmet-async';

import { configureAppStore } from 'store/configureStore';
import Template from '..';

const renderComponent = store =>
  render(
    <Provider store={store}>
      <HelmetProvider>
        <Template />
      </HelmetProvider>
    </Provider>,
  );

describe('<Template />', () => {
  let store;

  beforeEach(() => {
    store = configureAppStore();
  });
  it('should match the snapshot', () => {
    const component = renderComponent(store);
    expect(component.container.firstChild).toMatchSnapshot();
  });
});
