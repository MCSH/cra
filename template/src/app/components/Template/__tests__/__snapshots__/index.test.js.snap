// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Template /> should match the snapshot 1`] = `null`;
